addons/hr/models/hr_employee.py

```diff
@@ -252,10 +252,14 @@ def _copy_cache_from(self, public, field_names):
         # copy them to the cache of self; non-public data will be missing from
         # cache, and interpreted as an access error
         for fname in field_names:
-            values = self.env.cache.get_values(public, public._fields[fname])
-            if self._fields[fname].translate:
+            field = self._fields[fname]
+            public_field = public._fields[fname]
+            # no cache miss should be promised
+            values = self.env.cache.get_values(public_field, public.env.cache_key(public_field), public._ids)
```

Comment:
That complex call is awful.  We obviously need an extra method on fields for getting cache values
```suggestion
            values = public_field.get_cache_values(public)
```